### ISSUE TITLE
ARM Linux, .NET Core 2.0.4 crashes on shutdown

### ISSUE THREAD

**Issue by Const-me** [dotnet/coreclr#14997]
It happens after the control leaves my Program.Main method.

I’ve tried running my app with gdb, here’s the output:

```
Thread 14 "dotnet" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0xaa4cb450 (LWP 2405)]
0xb6781b9e in EventPipeConfiguration::RegisterProvider(EventPipeProvider&) ()
   from /opt/dotnet/shared/Microsoft.NETCore.App/2.0.4-servicing-25831-01/libcoreclr.so
```

Apparently, the previous fix doesn’t work: https://github.com/dotnet/coreclr/issues/13689

**Comment by jkotas** [comment]
cc @brianrob

**Comment by brianrob** [comment]
@Const-me, it looks like the fix for #13689 is not in 2.0.4.  I suspect you are hitting the same issue.  Can you see if the latest master branch build fixes this for you?  I can see about getting this pulled into a 2.0 servicing release.

**Comment by Const-me** [comment]
Thanks for the info. I will try tomorrow and let you know whether it works.

**Comment by Const-me** [comment]
@brianrob Yep, I can confirm it's fixed in 2.1.0-preview1-25912-02

**Comment by Imisnew2** [comment]
@Const-me We just upgraded to .NET Core 2.0.3 and we're also hitting this issue during CI, causing builds to fail intermittently. Has there been any update as to whether this is going to be put into a 2.0 servicing release per @brianrob on Nov 13?

I ask since the 2.0.X milestone doesn't have this (or a seemingly related issue) attached to it.

**Comment by brianrob** [comment]
@lmisnew2 I need to submit this for approval for 2.0 servicing.  I will do so and report back to this thread when I get a response.

**Comment by brianrob** [comment]
PR for 2.0: https://github.com/dotnet/coreclr/pull/15444

**Comment by Imisnew2** [comment]
Thanks @brianrob.

**Comment by brianrob** [comment]
Closing as this is fixed in master and will be merged into 2.0.
